### PR TITLE
remove duplicate conditions. add missing ;

### PR DESCRIPTION
**Pull request by a-tarasyuk**

### CODE REVIEW

File: src/native-common/View.tsx [hunk 1]
```diff
@@ -343,17 +343,20 @@ export class View extends ViewBase<Types.ViewProps, Types.Stateless> {
...
             accessibilityLiveRegion: AccessibilityUtil.accessibilityLiveRegionToString(props.accessibilityLiveRegion)
         };
-        if (_isNativeMacOs && App.supportsExperimentalKeyboardNavigation && (this.props.onPress ||
-                (this.props.tabIndex !== undefined && this.props.tabIndex >= 0))) {
+
+        if (_isNativeMacOs && App.supportsExperimentalKeyboardNavigation) {
             const macAccessibilityProps: MacComponentAccessibilityProps = accessibilityProps as any;
-            if (this.props.tabIndex !== -1) {
+
+            if (this.props.tabIndex && this.props.tabIndex >= 0) {
```
Comment (berickson1) [inline src/native-common/View.tsx]: This isn't correct. If tabIndex is 0, this condition will fail and the code below is not run
If tab index is undefined and onPress is defined, the code below should be run but is not

The conditional that was previously above is needed on this block, so I don't think it really optimizes anything by moving it

Comment (a-tarasyuk) [reply src/native-common/View.tsx]: ok, tabindex: -2

File: src/native-common/View.tsx [hunk 2]
```diff
@@ -343,17 +343,20 @@ export class View extends ViewBase<Types.ViewProps, Types.Stateless> {
...
             accessibilityLiveRegion: AccessibilityUtil.accessibilityLiveRegionToString(props.accessibilityLiveRegion)
         };
-        if (_isNativeMacOs && App.supportsExperimentalKeyboardNavigation && (this.props.onPress ||
-                (this.props.tabIndex !== undefined && this.props.tabIndex >= 0))) {
+
+        if (_isNativeMacOs && App.supportsExperimentalKeyboardNavigation) {
             const macAccessibilityProps: MacComponentAccessibilityProps = accessibilityProps as any;
-            if (this.props.tabIndex !== -1) {
+
+            if (this.props.tabIndex !== undefined && this.props.tabIndex >= 0) {
```
Comment (berickson1) [inline src/native-common/View.tsx]: This is still a change in behaviour.  If tabIndex is undefined and onPress is defined, the below code should be run.
You're going to end up with the same conditional that was above

**Comment by a-tarasyuk** [comment]
Oh, ok. For the first look, it is a little bit confused :). Have closed PR., In my opinion, this condition is too long and would be better to make it simpler or make a method with a readable name which describes the purpose of the condition. Seems case when tabIndex less than -1 causes the error...

**Pull request closed without merging**